crates/engine/tree/src/tree/payload_processor/multiproof.rs

```diff
@@ -385,91 +368,67 @@ impl MultiproofManager {
             return
         }
 
-        self.spawn_multiproof_task(input);
-    }
-
-    /// Spawns a multiproof task, dispatching to `spawn_storage_proof` if the input is a storage
-    /// multiproof, and dispatching to `spawn_multiproof` otherwise.
-    fn spawn_multiproof_task(&mut self, input: PendingMultiproofTask) {
         match input {
             PendingMultiproofTask::Storage(storage_input) => {
-                self.spawn_storage_proof(storage_input);
+                self.dispatch_storage_proof(storage_input);
             }
             PendingMultiproofTask::Regular(multiproof_input) => {
-                self.spawn_multiproof(multiproof_input);
+                self.dispatch_multiproof(multiproof_input);
             }
         }
     }
 
-    /// Spawns a single storage proof calculation task.
-    fn spawn_storage_proof(&mut self, storage_multiproof_input: StorageMultiproofInput) {
+    /// Dispatches a single storage proof calculation to worker pool.
+    fn dispatch_storage_proof(&mut self, storage_multiproof_input: StorageMultiproofInput) {
         let StorageMultiproofInput {
-            config,
-            source,
             hashed_state_update,
             hashed_address,
             proof_targets,
             proof_sequence_number,
-            state_root_message_sender,
             multi_added_removed_keys,
+            state_root_message_sender: _,
```

Comment:
yeah this feels a bit weird because we first clone stuff from self into the 

StorageMultiproofInput only to drop it here

imo this entire type StorageMultiproofInput is redundant